### PR TITLE
Ondemand compilation doesn't take scalacOptions

### PR DESCRIPTION
**Pull request by mlachkar**

### CODE REVIEW

**Comment by mlachkar** [comment]
linked to this https://github.com/scalameta/scalameta/pull/2094

**Comment by bjaglin** [comment]
Just landed on this and trying to understand the status: `scalacOptions` is actually passed but there is a bug in InteractiveSemanticdb for 2.13, right?

**Comment by mlachkar** [comment]
So if we pass the scalacOptions, it fixes the onDemand compilaton for 2.11 and 2.12, but not 2.13

**Comment by bjaglin** [comment]
Oh ok. I guess that's still something good to take in then (by removing tests for 2.13), and follow up 2.13 in another PR?

File: scalafix-cli/src/main/scala/scalafix/internal/v1/Args.scala [hunk 1]
```diff
@@ -376,11 +376,15 @@ case class Args(
...
-      settings.classpath.value = validatedClasspath.syntax
-      val reporter = new ConsoleReporter(settings)
-      LazyValue.fromUnsafe(() => new Global(settings, reporter))
+      LazyValue.fromUnsafe(() => {
+        val ss = new Settings()
+        val command = new CompilerCommand(scalacOptions, ss)
+        val settings = command.settings
+        settings.YpresentationAnyThread.value = true
+        settings.classpath.value = validatedClasspath.syntax
+        val reporter = new StoreReporter()
```
Comment (bjaglin) [inline scalafix-cli/src/main/scala/scalafix/internal/v1/Args.scala:385]: out of curiosity: is this block evaluated lazily on purpose to capture some state that it not available eagerly?

File: scalafix-tests/unit/src/test/scala/scalafix/tests/cli/CliSemanticSuite.scala [hunk 2]
```diff
@@ -257,4 +258,54 @@ class CliSemanticSuite extends BaseCliSuite {
     )
   }
 
+  test("on-demand with scalacOptions") {
+    // Todo: Ondemand compilation doesn't take scalacOptions or the reporter doesn't gather all warns in 2.13
+    // see https://github.com/scalacenter/scalafix/pull/1215
+    if (ScalaVersions.isScala213) true
```
Comment (bjaglin) [inline scalafix-tests/unit/src/test/scala/scalafix/tests/cli/CliSemanticSuite.scala:264]: should we open a separate issue to track that?

**Review by bjaglin — APPROVED** [review]
:+1: thanks for making this incremental!

File: scalafix-cli/src/main/scala/scalafix/internal/v1/Args.scala (hunk 1 above)
Comment (mlachkar) [reply scalafix-cli/src/main/scala/scalafix/internal/v1/Args.scala:385]: So the creation of Global was lazy, I just put the entire block as lazy. 
Usually, if the semanticdb file is present, global is never instantiated. 
I don't think there is any state that is captured later..

File: scalafix-tests/unit/src/test/scala/scalafix/tests/cli/CliSemanticSuite.scala (hunk 2 above)
Comment (mlachkar) [reply scalafix-tests/unit/src/test/scala/scalafix/tests/cli/CliSemanticSuite.scala:264]: ok

**Pull request merged**